File: Intermediate/Topic-004-Arrays-Prefix-Sum/Homework/Q1-Equilibrium-index-of-array/Equilibrium-index-of-array.py
class Solution:
    # @param A : list of integers
    # @return an integer
    def solve(self, A):
        total_sum = sum(A)
        left_sum=0
   
        for i in range(len(A)):
            right_sum = total_sum - left_sum - A[i]

            if right_sum == left_sum :
                return i
            left_sum += A[i]
        return -1

File: Intermediate/Topic-004-Arrays-Prefix-Sum/Homework/Q1-Equilibrium-index-of-array/Equilibrium-index-of-array.c
/**
 * @input A : Integer array
 * @input n1 : Integer array's ( A ) length
 * 
 * @Output Integer
 */
int solve(int* A, int n1) {
    int i, total, left_sum = 0;

    for (i = 0; i < n1; i++) {
        total += A[i];
    }

    for (i = 0; i < n1; i++) {
        int right_sum;
        right_sum = total - left_sum - A[i];

        if (right_sum == left_sum) {
            return i;
        }

        left_sum += A[i];
    }

    return -1;
}
